Add collapse styles to default.less

davidhatten opened the pull request:
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

[Related Issue](https://github.com/ant-design/ant-design/issues/9923)

This pushes color and padding for the Collapse component into defaults so they can easily be overridden.

I set up the values such that the Collapse will not change appearances for anyone currently using the defaults (or any values those defaults were using), but allows a user to remove all of the padding from both the header and the content of a Collapse.

Properties added to `default.less`:

```
@collapse-header-padding          
@collapse-header-bg
@collapse-content-padding         
@collapse-content-bg
```

I couldn't see any tests around default styles for any of the other components, but if these changes are testable I'll be happy to go back and add them. 

I don't think this requires any updates to documentation.

Extra checklist:

  ~~* [ ] Update API docs for the component.~~
  ~~* [ ] Update/Add demo to demonstrate new feature.~~
  ~~* [ ] Update TypeScript definition for the component.~~
  ~~* [ ] Add unit tests for the feature.~~

davidhatten committed:
push collapse styling into overridable defaults

afc163 commented on components/collapse/style/index.less:
```diff
@@ -3,9 +3,6 @@
 
 @collapse-prefix-cls: ~"@{ant-prefix}-collapse";
 
-@collapse-header-bg: @background-color-light;
```
You shouldn't change the original variable name cause someone may use it already.

davidhatten replied:
Good catch. Fixed a few lines below.

afc163 commented on components/style/themes/default.less:
```diff
@@ -461,3 +461,11 @@
+// Collapse
+// ---
+@collapse-header-padding:             12px 0 12px 40px;
+@collapse-header-background-color:    #fafafa;
```
#fafafa => @background-color-light

davidhatten committed:
Changes to ensure backwards compatibiliy, switch bg-color to be based on default style

afc163 commented on components/collapse/style/index.less:
```diff
@@ -3,8 +3,7 @@
 @collapse-prefix-cls: ~"@{ant-prefix}-collapse";
 
-@collapse-header-bg: @background-color-light;
-@collapse-active-bg: @background-color-base;
```
same as @collapse-active-bg

davidhatten replied:
@collapse-active-bg isn't used anywhere. Unless there's some kind of logic auto-attaching it to something, it looks like @collapse-active-bg is just cruft. I can restore it if you'd like, but it's not doing anything and it would be cleaner to remove it than leave it around looking like it's doing something.

davidhatten replied:
Unless it's *supposed* to be attached to something, but I don't have insight into that intent or why it's no longer attached.

afc163 replied:
It is Ok in that case.

afc163 commented on components/style/themes/default.less:
```diff
@@ -461,3 +461,11 @@
+// Collapse
+// ---
+@collapse-header-padding:             12px 0 12px 40px;
+@collapse-header-background-color:    @background-color-light;
```
use `@collapse-header-bg`

afc163 commented on components/style/themes/default.less:
```diff
@@ -461,3 +461,11 @@
+@collapse-header-padding:             12px 0 12px 40px;
+@collapse-header-background-color:    @background-color-light;
+@collapse-content-padding:            0 @padding-md;
+@collapse-content-background-color:   @component-background;
```
@collapse-content-bg

afc163 commented on components/style/themes/default.less:
```diff
@@ -461,3 +461,11 @@
+@collapse-header-background-color:    @background-color-light;
+@collapse-content-padding:            0 @padding-md;
+@collapse-content-background-color:   @component-background;
+@collapse-content-box-padding:        @padding-md 0 @padding-md 0;
```
`@collapse-content-box-padding` and `collapse-content-padding` are a little confused. Just use a `@collapse-padding: @padding-md` and use it in this way:

```
  & > &-box {
  padding-top: @collapse-padding;
  padding-bottom: @collapse-padding;
}
```

davidhatten committed:
change background-color to bg in style names

davidhatten committed:
simplify and consolodate content-padding

davidhatten committed:
I figured out why the styling is split across content and box like that

The collapse/show animation doesn't work right without the padding set up across the two elements like this. From an end-user point of view, I think this is the easiest thing.

davidhatten commented:
How about this suggestion?

It's a little weird, because the `@collapse-header-padding` gets to be a one-liner, but the `@collapse-content-padding-xxx` is enumerated. I can't see a good way around this. The structure needs to match the original like that in order for the animation to be smooth.

afc163 commented on components/style/themes/default.less:
```diff
@@ -461,3 +461,13 @@
+@collapse-content-padding-top:        @padding-md;
+@collapse-content-padding-right:      @padding-md;
+@collapse-content-padding-bottom:     @padding-md;
+@collapse-content-padding-left:       @padding-md;
```
Just @collapse-padding, I don't think we need to customize padding for each different direction.

davidhatten replied:
I would like that too, but see above comment.

If I don't put top/bottom padding on the inner `box` element inside `component`, the animation gets very chunky and gross.

davidhatten replied:
ah, but I didn't try putting all of the padding on the `box` instead of on the `component` until just now.

putting all the padding on the `box` works fine. 

Making this change now.

davidhatten committed:
pushing all padding styling on to the box instead of the content, keeping var name the same because the end user doesn't need to know the difference

afc163 commented:
The header background is different with demo now: https://ant.design/components/collapse/#header , check the `borderless` and `Custom Panel` demos.

davidhatten committed:
Pushing header-bg-color back up to the top of the component heirarchy

davidhatten commented:
There we go, I checked the whole page again, this one should now result in an identical docs page.

davidhatten committed:
shorten content padding

davidhatten commented:
Once again, I'm not sure why TravisCI is producing build errors. I'm correctly rebased off `feature/3.5.0`. Is there some kind of weird caching on TravisCI that's producing these errors? These build errors have nothing to do with the code I've touched.
